hazelcast/src/main/java/com/hazelcast/instance/NodeExtensionFactory.java

```diff
@@ -16,12 +16,14 @@
 
 package com.hazelcast.instance;
 
+import com.hazelcast.spi.annotation.PrivateApi;
 import com.hazelcast.util.ExceptionUtil;
 import com.hazelcast.util.ServiceLoader;
 
 import java.lang.reflect.Constructor;
 import java.util.Iterator;
 
+@PrivateApi
 public final class NodeExtensionFactory {
```

Comment:
Private API. It should not be used by anyone. If it would be something usable by an end user; we could not add new methods to it; e.g for hot restart.

Also NodeEngine exposes all kinds of implementation specifics like InternalSerializationService, IOService etc.